pkg/processor/aws_event_bridge.go

```diff
@@ -0,0 +1,189 @@
+	}
+	ruleARN = cfg.Options["aws_eventbridge_rule_arn"]
+
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region),
+		Credentials: credentials.NewStaticCredentials(
+			accessKey,
+			secretKey,
+			"", // a token will be created when the session it's used.
+		),
+	})
+	if err != nil {
+		return nil, errors.Errorf("could not create AWS session: %v", err)
+	}
+	eb.session = sess
+	ebSession := eventbridge.New(sess)
+	if ebSession == nil {
+		return nil, errors.Errorf("could not create AWS event bridge session")
+	}
+	eb.EventBridge = ebSession
+
+	rules, err := eb.ListRules(&eventbridge.ListRulesInput{})
+	if err != nil {
+		return nil, errors.Errorf("could not parse rules: %v", err)
+	}
+
+	// TODO: handle rules in non-default event bus
+	// TODO: handle NextToken() for paginated rule lists
+	for _, r := range rules.Rules {
+		log.Printf("found rule %s", *r.Arn)
+		switch {
+		case *r.Arn == ruleARN:
+			if r.EventPattern == nil {
+				return nil, errors.Errorf("rule event pattern must not be empty")
+			}
+
+			// log.Printf("found rule for vc-event-router: %s", *r.Arn)
+
+			var e eventPattern
+			err := json.Unmarshal([]byte(*r.EventPattern), &e)
+			if err != nil {
+				return nil, errors.Wrap(err, "could not parse event pattern:")
+			}
+
+			for _, s := range e.Detail.Subject {
+				log.Printf("adding event rule forwarding pattern %q to processor", s)
+				eb.patternMap[s] = *r.EventBusName
+			}
+			return &eb, nil
+		default:
+			continue
+		}
+
+	}
+	return nil, errors.Errorf("rule %s not found for configured AWS event bridge account", ruleARN)
+}
+
+// TODO: handle throttling/batching
+// TODO: break out into smaller funcs
+func (a *awsEventBridge) Process(moref types.ManagedObjectReference, baseEvent []types.BaseEvent) error {
+	// TODO: break this and loop out into helper
+	input := eventbridge.PutEventsInput{
+		Entries: []*eventbridge.PutEventsRequestEntry{},
+	}
+
+	for i, event := range baseEvent {
+		if a.verbose {
+			log.Printf("processing event [%d] from source %s: %+v", i, a.source, event)
+		}
+		eInfo := events.GetDetails(event)
+		a.mu.RLock()
+		defer a.mu.RUnlock() // TODO: finer grained locking, but fine for now since writes are not often
+
+		if _, ok := a.patternMap[eInfo.Name]; !ok {
+			// no event bridge rule pattern (subscription) for event, skip
+			continue
+		}
+		ce := events.NewCloudEvent(event, eInfo, a.source)
+		b, err := json.Marshal(ce)
+		if err != nil {
+			// TODO: skip or return err?
+			log.Printf("could not marshal cloud event, skipping: %v", err)
+			continue
+		}
+
+		bStr := string(b)
+		entry := eventbridge.PutEventsRequestEntry{
+			Detail:       aws.String(bStr),
+			EventBusName: aws.String(a.patternMap[eInfo.Name]),
+			Source:       aws.String(ce.Source),
+			DetailType:   aws.String(ce.Subject),
+		}
+		input.Entries = append(input.Entries, &entry)
+	}
+
+	// nothing to send
+	if len(input.Entries) == 0 {
+		return nil
+	}
+
+	r, err := a.PutEvents(&input)
```

Comment:
Does `r` refer to response? if so can we name it to `eventResponse`

Reply:
I like `res`. It's short enough to indicate being of less importance, and easy to carry in human memory for a few lines. I think `r` can be confused for request, perhaps.

Reply:
`resp`